src/include/migraphx/matcher.hpp

```diff
@@ -415,7 +418,16 @@ void find_matches_for(source_location location, Mod& mod, instruction_ref ins, M
                 return;
             if(trace > 1 and trace_for)
                 std::cout << "Match: " << matcher_name << std::endl;
+
+            timer match_timer;            
             auto r = match_instruction(get_module(mod), ins, m.matcher());
+            auto match_time = match_timer.record<std::chrono::duration<double, std::micro>>();
```

Comment:
We can't directly use the existing time function from time.hpp for timing the match_instruction because it only records the time and doesn't return any value. Since match_instruction needs to assign its result to r, we would require a separate time template function that can both capture the result and measure the elapsed time as overloading is not possible as it is leading to ambiguity in function selection. This additional template would return a std::pair, containing both the result and the time, Any thoughts on this @pfultz2 ?